client/my-sites/checkout/checkout-thank-you/domain-mapping-details.jsx

```diff
@@ -10,128 +10,128 @@ import { localize } from 'i18n-calypso';
-						"{{registrarSupportLink}}domain provider's site{{/registrarSupportLink}} " +
-						'(where you purchased the domain), and update your name servers to:',
-					{
-						components: {
-							registrarSupportLink: registrarSupportLink,
-						},
-					}
-				) }
-			</p>
-			<ul className="checkout-thank-you__domain-mapping-details-nameservers">
-				<li>ns1.wordpress.com</li>
-				<li>ns2.wordpress.com</li>
-				<li>ns3.wordpress.com</li>
-			</ul>
-		</div>
+	const primaryMessage = translate(
+		'Please log into your account at your domain registrar and {{strong}}update the name servers{{/strong}} of your domain to use the following values, as detailed in {{link}}these instructions{{/link}}:',
+		{
+			comment: 'Notice for mapped domain notice with NS records pointing to somewhere else',
+			components: {
+				strong: <strong />,
+				link: renderLinkTo( MAP_DOMAIN_CHANGE_NAME_SERVERS ),
+			},
+		}
 	);
 
-	if ( isSubdomainMapping ) {
-		instructions = (
-			<div>
-				<p>
-					{ translate(
-						'To point your domain at your WordPress.com site, log in to your ' +
-							"{{registrarSupportLink}}domain provider's site{{/registrarSupportLink}} " +
-							'(where you purchased the domain), and edit the DNS records to add a CNAME record:',
-						{
-							components: {
-								registrarSupportLink: registrarSupportLink,
-							},
-						}
-					) }
-				</p>
-				<ul className="checkout-thank-you__domain-mapping-details-nameservers">
-					<li>
-						{ domain }. IN CNAME { selectedSiteDomain }.
-					</li>
-				</ul>
-			</div>
+	const renderRecommendedSetupMessage = () => {
+		return (
+			<FoldableFAQ
+				id="recommended-mapping-setup"
+				question={ translate( 'Recommended setup' ) }
+				expanded
+			>
+				<p>{ primaryMessage }</p>
+				{ ! isSubdomain( domain.name ) && (
+					<ul className="checkout-thank-you__name-server-list">
+						{ WPCOM_DEFAULT_NAMESERVERS.map( ( nameServer ) => {
+							return <li key={ nameServer }>{ nameServer }</li>;
+						} ) }
+					</ul>
+				) }
+			</FoldableFAQ>
 		);
-	}
+	};
 
-	if ( isSubdomainMapping && isBusinessPlan ) {
-		instructions = (
-			<div>
-				<p>
-					{ translate(
-						'To point your domain at your WordPress.com site, log in to your ' +
-							"{{registrarSupportLink}}domain provider's site{{/registrarSupportLink}} " +
-							'(where you purchased the domain), and edit the DNS records to add these NS records:',
-						{
-							components: {
-								registrarSupportLink: registrarSupportLink,
-							},
-						}
-					) }
-				</p>
-				<ul className="checkout-thank-you__domain-mapping-details-nameservers">
-					<li>{ domain }. IN NS ns1.wordpress.com.</li>
-					<li>{ domain }. IN NS ns2.wordpress.com.</li>
-					<li>{ domain }. IN NS ns3.wordpress.com.</li>
+	const isDataLoaded = () => {
+		return getSelectedDomain( { domains, selectedDomainName: domain } );
+	};
+
+	const renderARecordsList = () => {
+		if ( isDataLoaded() ) {
```

Comment:
Maybe let's inverse the logic here - and return the placeholder component early. As that's the least interesting case (and avoids big nested lines).